sdk/core/src/bytes_response.rs

```diff
@@ -138,3 +40,43 @@ impl BytesResponse {
         (self.status, self.headers, self.body)
     }
 }
+
+#[derive(Serialize, Deserialize)]
+pub(crate) struct SerializedBytesResponse {
+    status: u16,
+    headers: HashMap<String, String>,
+    body: String,
+}
+
+impl From<BytesResponse> for SerializedBytesResponse {
+    fn from(r: BytesResponse) -> Self {
+        let mut headers = HashMap::new();
+        for (h, v) in r.headers.iter() {
+            headers.insert(h.as_str().into(), v.to_str().unwrap().into());
+        }
+        let status = r.status.as_u16();
+        let body = base64::encode(&r.body as &[u8]);
+        Self {
+            status,
+            headers,
+            body,
+        }
+    }
+}
```

Comment:
1.) Efficiency is pretty much the same. The only difference is that part of the "deserialization" has been moved to a second function. This might mean a marginally slower process if the function isn't inlined but we're talking on the order of a few instructions. I don't think it will make any perceptible difference. 
2.) I agree it's a bit of a shame that you have to do the two step process manually. I'll fix this.
3.) I don't understand this point?
4.) The serialization code is 40% smaller - that seems like a big win for maintenance.